src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -299,14 +301,26 @@ public void addTier(Tier tier, ArrayList<String> roomNums) {
 
     }
 
-
+    //=========== Billing System =============================================================================
 
     @Override
     public void setRoomCost(Room room, RoomCost roomCost) {
         requireAllNonNull(room, roomCost);
         room.setCost(roomCost);
     }
 
+    @Override
+    public void addService(PersonId personId, RoomId roomId, Service service) {
+        requireAllNonNull(personId, roomId, service);
+        ArrayList<Bill> bills = addressBook.findPersonWithId(personId).get().getBills();
+
+        for (Bill bill : bills) {
+            if (bill.getRoomId().equals(roomId)) {
+                bill.addChargeable(service);
+            }
+        }
+    }
```

Comment:
Do you think it's better to put this in Person class? So here, just check whether the person exists.